pkg/sdkprovider/sdk_provider.go

```diff
@@ -0,0 +1,46 @@
+package sdkprovider
+
+import (
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/client/v1/nimbleos"
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/param"
+)
+
+// VolumeService defines the interface to Volume related operations
+type VolumeService interface {
+	GetVolumes(params *param.GetParams) ([]*nimbleos.Volume, error)
+	CreateVolume(obj *nimbleos.Volume) (*nimbleos.Volume, error)
+	UpdateVolume(id string, obj *nimbleos.Volume) (*nimbleos.Volume, error)
+	GetVolumeById(id string) (*nimbleos.Volume, error)
+	GetVolumeByName(name string) (*nimbleos.Volume, error)
+	GetVolumeBySerialNumber(serialNumber string) (*nimbleos.Volume, error)
+	OnlineVolume(id string, force bool) (*nimbleos.Volume, error)
+	OfflineVolume(id string, force bool) (*nimbleos.Volume, error)
+	DeleteVolume(id string) error
+	AssociateVolume(volId string, volcollId string) error
+	DisassociateVolume(volId string) error
+	RestoreVolume(id string, baseSnapId string) error
+	MoveVolume(id string, destPoolId string, forceVvol *bool) (*nimbleos.NsVolumeListReturn, error)
+	BulkMoveVolumes(volIds []*string, destPoolId string, forceVvol *bool) (*nimbleos.NsVolumeListReturn, error)
+	AbortMoveVolume(id string) error
+	BulkSetDedupeVolumes(volIds []*string, dedupeEnabled bool) error
+	BulkSetOnlineAndOfflineVolumes(volIds []*string, online bool) error
+	DestroyVolume(id string) error
```

Comment:
We must implement this in the VolumeService.go as well otherwise it complains that VolumeService is not implementing the sdkprovider.volumeservice.

Reply:
# github.hpe.com/nimble-dcs/golang-sdk/pkg/service
/Users/kallianc/go/pkg/mod/github.hpe.com/nimble-dcs/golang-sdk@v0.0.0-20201110084753-b00225cd1877/pkg/service/group_service_factory.go:170:20: cannot use NewVolumeService(gs) (type *VolumeService) as type sdkprovider.VolumeService in assignment:
	*VolumeService does not implement sdkprovider.VolumeService (missing DestroyVolume method)

Reply:
You are checking in private GitHub where it doesnt exist . You should check in public git repo . I have already validated this and it works. 
In private repo this code is not checked in yet . it will be done after public git PR is approved.

Reply:
thanks. I see DestroyVolume. Since array-exporter was using private github, I copied all changes to that.